todoist/managers/archive.py

```diff
@@ -0,0 +1,188 @@
+"""
+Managers to get the list of archived items and sections.
+
+Manager makers available as "items_archive" and "sections_archive" attributes of
+API object.
+
+
+Usage example (for items).
+
+```python
+
+# Create an API object
+import todoist
+api = todoist.TodoistAPI(...)
+
+# Get project ID (take inbox)
+project_id = api.user.get()['inbox_project']
+
+# Initiate ItemsArchiveManager
+archive = api.items_archive.for_project(project_id)
+
+# Iterate over the list of completed items for the archive
+for item in archive.items():
+    print(item["date_completed"], item["content"])
+```
+"""
+
+from typing import TYPE_CHECKING
+
+from ..models import Item, Model, Section
+from .generic import Manager
+
+if TYPE_CHECKING:
+    from ..api import TodoistAPI
+
+
+class ArchiveManager(Manager):
+
+    object_model = Model
+
+    def __init__(self, api, object_type):
+        # type: (TodoistAPI, str) -> None
+        assert object_type in {"sections", "items"}
+        self.api = api
+        self.cursor = None
+        self.has_more = True
+        self.elements = []
+        self.object_type = object_type
+
+    def reset(self):
+        """Reset internal cache of manager and start iteration from scratch."""
+        self.has_more = True
+        self.cursor = None
+        self.elements[:] = []
+
+    def next_page(self):
+        """Return response for the next page of the archive."""
+        resp = self.api.session.get(
+            self._next_url(),
+            params=self._next_query_params(),
+            headers=self._request_headers(),
+        )
+        resp.raise_for_status()
+        return resp.json()
+
+    def _next_url(self):
+        return "{0}/sync/{1}/archive/{2}".format(
+            self.api.api_endpoint, self.api.api_version, self.object_type
+        )
+
+    def _next_query_params(self):
+        ret = {}
+        if self.cursor:
+            ret["cursor"] = self.cursor
+        return ret
+
+    def _request_headers(self):
+        return {"Authorization": "Bearer {}".format(self.api.token)}
+
+    def _iterate(self):
+        for el in self.elements:
+            yield el
+
+        while True:
+            if not self.has_more:
+                break
+
+            resp = self.next_page()
+            objects = resp[self.object_type]
+
+            self.elements += [self._make_object(data) for data in objects]
+            self.has_more = resp["has_more"]
+            self.cursor = resp.get("next_cursor")
+
+            for obj in objects:
+                yield obj
+
+    def _make_object(self, data):
+        return self.object_model(data, self.api)
+
+
+class SectionsArchiveManagerMaker(object):
+    def __init__(self, api):
+        self.api = api
```

Comment:
Same as [comment above](https://github.com/Doist/todoist-python/pull/68/files#r352471364) about `self.api`.

Reply:
This is not a subclass of Manager, so this element is necessary.